### PR TITLE
Add boilerplate catalog-info.yaml and pipeline.yml file

### PR DESCRIPTION
**Pull request by kvalliyurnatt**
Adding initial catalog-info.yaml file to register repository with buildkite
Part of: [CP-2778](https://elasticco.atlassian.net/browse/CP-2778)

### CODE REVIEW

File: catalog-info.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  implementation:
+    apiVersion: buildkite.elastic.dev/v1
+    kind: Pipeline
+    metadata:
+      name: elasticsearch-k8s-metrics-adapter
+    spec:
+      repository: elastic/elasticsearch-k8s-metrics-adapter
+      pipeline_file: ".buildkite/pipeline.yml"
+      teams:
+        cloud-k8s-operator: {}
```
Comment (thbkrkr) [inline catalog-info.yaml:25]: We probably want the same config than https://github.com/elastic/lightweight-cert-manager/blob/15a35783b13d1a7b54aa901c65e931b4246cacd6/catalog-info.yaml#L47-L49:

```suggestion
        cloud-k8s-region:
          access_level: MANAGE_BUILD_AND_READ
```

File: catalog-info.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  type: buildkite-pipeline
+  owner: group:cloud-k8s-operator
+  system: buildkite
+  implementation:
+    apiVersion: buildkite.elastic.dev/v1
+    kind: Pipeline
+    metadata:
+      name: elasticsearch-k8s-metrics-adapter
+    spec:
+      repository: elastic/elasticsearch-k8s-metrics-adapter
```
Comment (thbkrkr) [inline catalog-info.yaml:22]: FYI defaults values for the `provider_settings` are defined [here](https://github.com/elastic/ci/blob/685b84f1545ac68c3553a1316d5ef2da4a8cc1a3/terrazzo/terrazzo/config/defaults/buildkite/pipelines.py#L4), which should be ok.

You can look at the settings of another serverless project via the ui to compare: https://buildkite.com/elastic/elasticsearch-controller/settings/repository.

File: catalog-info.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      pipeline_file: ".buildkite/pipeline.yml"
```
Comment (thbkrkr) [inline catalog-info.yaml:23]: You could add a simple `.buildkite/pipeline.yml` if you want to see the first builds green.

```sh
> echo 'steps:
  - command: echo "🐙"' > .buildkite/pipeline.yml
```

File: catalog-info.yaml (hunk 2 above)
Comment (kvalliyurnatt) [reply catalog-info.yaml:22]: I think the defaults look fine for now, will tweak in the next PR as we add actual things to the pipeline

**Review by thbkrkr — APPROVED** [review]

**Pull request merged**